x/ecocredit/server/core/create_class.go

```diff
@@ -0,0 +1,124 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// CreateClass creates a new class of ecocredit
+//
+// The admin is charged a fee for creating the class. This is controlled by
+// the global parameter CreditClassFee, which can be updated through the
+// governance process.
+func (k Keeper) CreateClass(goCtx context.Context, req *v1beta1.MsgCreateClass) (*v1beta1.MsgCreateClassResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(goCtx)
+	adminAddress, err := sdk.AccAddressFromBech32(req.Admin)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: remove params https://github.com/regen-network/regen-ledger/issues/729
+	var params ecocredit.Params
+	k.params.GetParamSet(sdkCtx, &params)
+	if params.AllowlistEnabled && !k.isCreatorAllowListed(sdkCtx, params.AllowedClassCreators, adminAddress) {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("%s is not allowed to create credit classes", adminAddress.String())
+	}
+
+	// Charge the admin a fee to create the credit class
+	err = k.chargeCreditClassFee(sdkCtx, adminAddress, params.CreditClassFee)
+	if err != nil {
+		return nil, err
+	}
+
+	creditType, err := k.getCreditType(req.CreditTypeAbbrev, params.CreditTypes)
+	if err != nil {
+		return nil, err
+	}
+
+	// default the sequence to 1 for the `not found` case.
+	// will get overwritten by the actual sequence if it exists.
+	var seq uint64 = 1
+	classSeq, err := k.stateStore.ClassSequenceStore().Get(goCtx, creditType.Abbreviation)
+	if err != nil {
+		if !ormerrors.IsNotFound(err) {
```

Comment:
We can use &&

Reply:
that would trigger the else case, which would read from the pointer even on `NotFound` errors